### PR TITLE
[FIX] point_of_sale: disable payment editing for printed orders

### PR DESCRIPTION
**Pull request by davidmonnom**
Before this commit, it was possible to edit the payment lines of a printed order.

This commit disables the editing of payment lines for printed orders.

taskId: 4178904

### CODE REVIEW

File: addons/point_of_sale/static/src/app/store/pos_store.js [hunk 1]
```diff
@@ -1355,18 +1355,21 @@ export class PosStore extends Reactive {
         const baseUrl = this.session._base_url;
         return order.export_for_printing(baseUrl, headerData);
     }
-    async printReceipt() {
+    async printReceipt(order = this.get_order()) {
+        const nbrPrint = order.nb_print;
+        await this.data.write("pos.order", [order.id], { nb_print: nbrPrint + 1 });
```
Comment (caburj) [inline addons/point_of_sale/static/src/app/store/pos_store.js:1360]: Shouldn't we write the value after it successfully printed?

File: addons/point_of_sale/static/src/app/screens/ticket_screen/reprint_receipt_button/reprint_receipt_button.js
Comment (caburj) [inline addons/point_of_sale/static/src/app/screens/ticket_screen/reprint_receipt_button/reprint_receipt_button.js:1]: Normally, when there is no printer setup in the config, a screen is displayed showing the receipt. Why are we removing it? There is no mention of this removal from the spec.

Comment (caburj) [reply addons/point_of_sale/static/src/app/screens/ticket_screen/reprint_receipt_button/reprint_receipt_button.js:1]: Got answer from MOBT. He agreed with this changed already.

File: addons/point_of_sale/static/src/app/store/pos_store.js (hunk 1 above)
Comment (davidmonnom) [reply addons/point_of_sale/static/src/app/store/pos_store.js:1360]: Yep

**Comment by caburj** [comment]
robodoo r+

**Comment by xiaohao0576** [comment]
This modification has caused a lot of inconvenience to the restaurant's billing process. If I print out the bill in advance and the customer finds an error, and it is indeed the restaurant staff who has selected the wrong price, is it not allowed to modify the price in such a case? Mya I know what is the correct way to handle such a situation if it does occur?

**Comment by xiaohao0576** [comment]
> This modification has caused a lot of inconvenience to the restaurant's billing process. If I print out the bill in advance and the customer finds an error, and it is indeed the restaurant staff who has selected the wrong price, is it not allowed to modify the price in such a case? Mya I know what is the correct way to handle such a situation if it does occur?

This bug fixed, please check link blow:
https://github.com/odoo/odoo/pull/182576

**Pull request closed without merging**